### DISCUSSION TITLE
Run a post-entrypoint script on build cancellation

### DISCUSSION THREAD

**Discussion by obrassard** [community/community#25987] [Actions]
<p>Hello everyone ! I am working on a Docker based Action and I am wondering if there is a way to run a post-entrypoint cleanup script whenever a build is cancelled by a user.</p>
<p>More specifically, will <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post-entrypoint" rel="nofollow noopener"><code>post-entrypoint</code></a> run even if a build is cancelled, and is there a way to know in the runners contexts that a build is getting canceled so I could use this condition with <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post-if" rel="nofollow noopener"> <code>post-if</code> </a>?</p>
<p>Thanks in advance !</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<aside class="quote no-group quote-modified" data-username="obrassard" data-post="1" data-topic="120845">
<div class="title">
<div class="quote-controls"></div>
 obrassard:</div>
<blockquote>
<p>More specifically, will <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post-entrypoint" rel="nofollow noopener"> <code>post-entrypoint</code> </a> run even if a build is cancelled?</p>
</blockquote>
</aside>
<p>Yes, post-entrypoint will always run in default even if a build is cancelled.<br>
<img src="https://user-images.githubusercontent.com/24806478/181094434-853d33e3-df76-4487-830c-cc8cbd39c7ed.png" alt="post action" data-base62-sha1="4KwscNuc53QpJ7Bt5NuPQqdJyg2" width="659" height="372"><br>
The  <code>post:</code>  action will only run if the conditions in  <code>post-if</code>  are met. If <code>post-if</code> is not set, then the condition defaults to  <code>always()</code> .<br>
You could use <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#cancelled" rel="nofollow noopener">cancelled()</a> function in post-if, then the post action will only run when the build is canceled.<br>
<img src="https://user-images.githubusercontent.com/24806478/181094435-92e910d8-f445-4615-8fe9-210d0c1058c9.png" alt="if canceled" data-base62-sha1="ycefCj4spBWncNXM1fAoHcQryEF" width="662" height="364"></p>

**Comment by obrassard** [comment]
<p>Thanks a lot <a class="mention" href="/yanjingzhu">@yanjingzhu</a> for your answer !</p>

**Comment by obrassard** [comment]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, I tried to use the <code>cancelled()</code> function with post-if (in my action.yml) but after testing, I realize that it doesn’t work:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a397af28029b3d8d6d1d89a1fe722d2c7d020dcf.png" data-download-href="/uploads/short-url/nlcOc4rDZVWTg7Ti4mxnuwzsf9l.png?dl=1" title="Capture d’écran, le 2020-07-02 à 13.43.08" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/17227551/181094439-f7585ae2-2a91-427f-be5f-6561eea8431c.png" alt="Capture d’écran, le 2020-07-02 à 13.43.08" data-base62-sha1="nlcOc4rDZVWTg7Ti4mxnuwzsf9l" width="690" height="212" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a397af28029b3d8d6d1d89a1fe722d2c7d020dcf_2_690x212.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a397af28029b3d8d6d1d89a1fe722d2c7d020dcf_2_1035x318.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/a/a397af28029b3d8d6d1d89a1fe722d2c7d020dcf_2_1380x424.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a397af28029b3d8d6d1d89a1fe722d2c7d020dcf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture d’écran, le 2020-07-02 à 13.43.08</span><span class="informations">1708×526 54.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here an excerpt of my action.yml file :slight_smile:</p>
<pre><code class="lang-yaml">runs:
  using: 'docker'
  image: 'Dockerfile'
  entrypoint: '/app/entrypoint.sh'
  post-entrypoint: '/app/dokku-unlock.sh'
  post-if: ${{ cancelled() }}
</code></pre>
<p>Am i missing something ?<br>
Thanks a lot in advance !</p>
<p>UPDATE : Never mind, using <code>post-if: cancelled()</code> (without <code>${{ }}</code>) fix the issue</p>
